services/payments/src/simcore_service_payments/services/notifier_email.py

```diff
@@ -123,6 +130,22 @@ class _PaymentData:
     price_dollars: str
     osparc_credits: str
     invoice_url: str
+    invoice_pdf: str
+
+
+transient_errors = (httpx.HTTPStatusError, httpx.ConnectError, httpx.ReadTimeout)
```

Comment:
THOUGHT: interesting selection of exceptions. In the case of status error, shouldnt' you retry only for specific status codes? E.g. imaging you are getting a 404 because the link is wrong

Reply:
Improved:
```
def retry_if_status_code(response):
    return response.status_code in (429, 500, 502, 503, 504)  # Retry for these common transient errors

exception_retry_condition = retry_if_exception_type((httpx.ConnectError, httpx.ReadTimeout))
result_retry_condition = retry_if_result(retry_if_status_code)

@retry(
    retry=exception_retry_condition | result_retry_condition,
    wait=wait_exponential(multiplier=1, min=4, max=10),
    stop=stop_after_attempt(5),
    retry_error_callback=lambda _: None,  # Return None if all retries fail
)
```